server/pulp/server/async/tasks.py

```diff
@@ -108,17 +171,16 @@ def _delete_worker(name, normal_shutdown=False):
         _logger.error(msg)
 
     # Delete the worker document
-    worker_list = list(resources.filter_workers(Criteria(filters={'_id': name})))
-    if len(worker_list) > 0:
-        worker_document = worker_list[0]
-        worker_document.delete()
+    worker = Worker.objects(name=name).first()
```

Comment:
I think you can just call ``.delete()`` instead of ``.first()``, and ditch the lines below.